action.yml

```diff
@@ -39,28 +40,67 @@ runs:
         registry: ghcr.io
         username: ${{ inputs.github_username }}
         password: ${{ inputs.github_token }}
-    - name: Check that the tag exists
-      run: docker manifest inspect ghcr.io/${{ inputs.image }} > /dev/null
+    - name: Prepend ghcr.io to image name if necessary
+      id: image_name
+      uses: actions/github-script@v6
+      env:
+        IMAGE: ${{ inputs.image }}
+      with:
+        script: |
+          image = process.env.IMAGE
+          if(!image.startsWith('ghcr.io/')) {
+            image = "ghcr.io/" + image
+          }
+          core.setOutput('image',image)
+    - name: Check that the image & tag exist
+      run: docker manifest inspect "$IMAGE" > /dev/null
       shell: bash
+      env:
+        IMAGE: ${{ steps.image_name.outputs.image }}
     - uses: azure/setup-kubectl@v1
     - name: Authenticate with kubernetes
       shell: bash
       run: |
+        set -x
         mkdir -p ${HOME}/.kube/certs/cluster
         echo ${{ inputs.cluster_ca }} | base64 -d > ${HOME}/.kube/certs/cluster/k8s-ca.crt
-        kubectl config set-cluster cluster --certificate-authority=${HOME}/.kube/certs/cluster/k8s-ca.crt --server=${{ inputs.cluster_server }}
-        kubectl config set-credentials default --token=`echo ${{ inputs.namespace_token }} | base64 -d`
-        kubectl config set-context default --cluster=cluster --user=default --namespace=${{ inputs.namespace }}
+        kubectl config set-cluster cluster --certificate-authority=${HOME}/.kube/certs/cluster/k8s-ca.crt --server="$CLUSTER_SERVER"
+        kubectl config set-credentials default --token=$(echo "$NAMESPACE_TOKEN" | base64 -d)
+        kubectl config set-context default --cluster=cluster --user=default --namespace="$NAMESPACE"
         kubectl config use-context default
+      env:
+        CLUSTER_CA: ${{ inputs.cluster_ca }}
+        CLUSTER_SERVER: ${{ inputs.cluster_server }}
+        NAMESPACE_TOKEN: ${{ inputs.namespace_token }}
+        NAMESPACE: ${{ inputs.namespace }}
     - name: Deploy
       shell: bash
       run: |
-        if echo ${{ inputs.type }} | grep -c "deployment"
-        then
-          kubectl set image deployment ${{ inputs.deployment }} ${{ inputs.container }}=ghcr.io/${{ inputs.image }}
-        elif echo ${{ inputs.type }} | grep -c "cronjob"
-        then
-          kubectl patch cronjob ${{ inputs.cronjob_name }} --type=json -p='[{"op":"replace", "path": "/spec/jobTemplate/spec/template/spec/containers/0/image", "value":"ghcr.io/${{ inputs.image }}"}]'
-        else
-          echo "type must be deployment or cronjob"
-        fi
+        set -x
+        case "$TYPE" in
```

Comment:
I assume this is because `case` is nicer than `if/else`?

Reply:
:+1: This is a cleaner way to do it IMO.